server/handler.go

```diff
@@ -518,11 +523,13 @@ func (h *Handler) AddTransferRegionOperator(regionID uint64, storeIDs map[uint64
 	}
 
 	peers := make(map[uint64]*metapb.Peer)
-	for id := range storeIDs {
-		peers[id] = &metapb.Peer{StoreId: id}
+	roles := make(map[uint64]placement.PeerRoleType)
+	for id, role := range storeIDsAndRoles {
+		peers[id] = &metapb.Peer{StoreId: id, Role: role.MetaPeerRole()}
+		roles[id] = role
 	}
 
-	op, err := operator.CreateMoveRegionOperator("admin-move-region", c, region, operator.OpAdmin, peers)
+	op, err := operator.CreateMoveRegionOperator("admin-move-region", c, region, operator.OpAdmin, peers, roles)
```

Comment:
I think we can add some check to make sure there is at most 1 "leader" and there are at least 1 "leader" or "voter".

Reply:
Sounds good! Check added.